src/Avalonia.Controls/Utils/UndoRedoHelper.cs

```diff
@@ -8,44 +8,41 @@
 
 namespace Avalonia.Controls.Utils
 {
-    class UndoRedoHelper<TState> : WeakTimer.IWeakTimerSubscriber where TState : IEquatable<TState>
+    class UndoRedoHelper<TState> : WeakTimer.IWeakTimerSubscriber where TState : struct, IEquatable<TState>
     {
         private readonly IUndoRedoHost _host;
 
         public interface IUndoRedoHost
         {
-             TState UndoRedoState { get; set; }
+            TState UndoRedoState { get; set; }
         }
-        
+
 
 
         private readonly LinkedList<TState> _states = new LinkedList<TState>();
 
-        [NotNull]
         private LinkedListNode<TState> _currentNode;
 
         public int Limit { get; set; } = 10;
 
         public UndoRedoHelper(IUndoRedoHost host)
         {
             _host = host;
-            _states.AddFirst(_host.UndoRedoState);
-            _currentNode = _states.First;
-            WeakTimer.StartWeakTimer(this, new TimeSpan(0, 0, 1));
-
+            //_states.AddFirst(_host.UndoRedoState);
+            //_currentNode = _states.First;
```

Comment:
Could you remove the commented-out code? I don't like commented-out in general as it's difficult to tell why it was commented out when we come back to it in a year's time.